src/Console/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,205 @@
+    /** @internal */
+    const PACKAGIST_PACKAGE_NAME = 'humbug/php-scoper';
+    /** @internal */
+    const ROLLBACK_OPT = 'rollback';
+    /** @internal */
+    const CHECK_OPT = 'check';
+
+    /**
+     * @var OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var string
+     */
+    private $version;
+
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setDescription('Update php-scoper.phar to most recent stable build.')
+            ->addOption(
+                self::ROLLBACK_OPT,
+                'r',
+                InputOption::VALUE_NONE,
+                'Rollback to previous version of php-scoper if available on filesystem.'
+            )
+            ->addOption(
+                self::CHECK_OPT,
+                'c',
+                InputOption::VALUE_NONE,
+                'Checks whether an update is available.'
+            )
+        ;
+    }
+
+    /**
+     * @param InputInterface  $input
+     * @param OutputInterface $output
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->output = $output;
+        $this->version = $this->getApplication()->getVersion();
+
+        if ($input->getOption('rollback')) {
+            $this->rollback();
+
+            return;
+        }
+
+        if ($input->getOption('check')) {
+            $this->printAvailableUpdates();
+
+            return;
+        }
+
+        $this->updateToStableBuild();
+    }
+
+    private function updateToStableBuild()
+    {
+        $this->update($this->getStableUpdater());
+    }
+
+    /**
+     * @return Updater
+     */
+    private function getStableUpdater(): Updater
+    {
+        $updater = new Updater();
+        $updater->setStrategy(Updater::STRATEGY_GITHUB);
+        return $this->getGithubReleasesUpdater($updater);
+    }
+
+    private function update(Updater $updater)
+    {
+        $this->output->writeln('Updating...'.PHP_EOL);
+        try {
+            $result = $updater->update();
+
+            $newVersion = $updater->getNewVersion();
+            $oldVersion = $updater->getOldVersion();
+        
+            if ($result) {
+                $this->output->writeln('php-scoper has been updated.');
+                $this->output->writeln(sprintf(
+                    'Current version is: %s.',
+                    $newVersion
+                ));
+                $this->output->writeln(sprintf(
+                    'Previous version was: %s.',
+                    $oldVersion
+                ));
+            } else {
+                $this->output->writeln('php-scoper is currently up to date.');
+                $this->output->writeln(sprintf(
+                    'Current version is: %s.',
+                    $oldVersion
+                ));
+            }
+        } catch (\Exception $e) {
+            $this->output->writeln(sprintf('Error: %s', $e->getMessage()));
```

Comment:
The basic update flow is to backup the original phar (leaving original still in place), download the new phar to a temporary directory, validate it, and then copy it to the original phar location (overwrite). On error, the temporary copy would be deleted. The rollback is largely in the event that an update is valid, but broken or incompatible for any reason and you want the old version back.